lib/util/LocationStatusManager.js

```diff
@@ -59,11 +203,476 @@ class LocationStatusManager {
+     */
+    async _executeScheduleResumeOperation(location, service, options, stateMachine) {
+        const paused = this._locationStatusStore[location].getServicePauseStatus(service);
+        if (paused) {
+            const scheduleDate = options.scheduleDate;
+            this._locationStatusStore[location].resumeLocation(service, scheduleDate);
+            stateMachine.executeStateTransition(OPERATION_TYPES.SCHEDULE_RESUME, { scheduleDate });
+            
+            try {
+                await this._updateServiceStatusForLocationAsync(location);
+                this._scheduleResumeJob(location, service, scheduleDate);
+            } catch (error) {
+                this._locationStatusStore[location].setServiceResumeSchedule(service, null);
+                stateMachine.executeStateTransition(OPERATION_TYPES.PAUSE);
+                throw error;
+            }
+        }
+    }
+
+    /**
+     * Execute delete schedule operation
+     */
+    async _executeDeleteScheduleOperation(location, service, stateMachine) {
+        const paused = this._locationStatusStore[location].getServicePauseStatus(service);
+        if (paused) {
+            this._deleteResumeJob(location, service);
+            this._locationStatusStore[location].setServiceResumeSchedule(service, null);
+            stateMachine.executeStateTransition(OPERATION_TYPES.DELETE_SCHEDULE);
+            await this._updateServiceStatusForLocationAsync(location);
+        }
+    }
+
+    /**
+     * Convert callback-based update to promise
+     */
+    _updateServiceStatusForLocationAsync(location) {
+        return promisify(this._updateServiceStatusForLocation.bind(this))(location);
     }
 
     /**
-     * Creates mongo collection for
-     * storing the location status
+     * Queue an operation for processing with configurable timeout
+     */
+    _queueOperation(type, service, locations, options = {}) {
+        return new Promise((resolve, reject) => {
+            const operations = locations.map(location => {
+                const operationId = this._generateOperationId();
+                const operation = new ServiceLocationOperation(operationId, type, service, location, options);
+                this._pendingOperations.set(operationId, operation);
+                return operation;
+            });
+
+            const startTime = Date.now();
+
+            // Cleanup function
+            const cleanup = () => {
+                operations.forEach(op => {
+                    this._pendingOperations.delete(op.id);
+                });
+            };
+
+            operations.forEach(operation => {
+                this._operationQueue.push(operation);
+            });
+            const checkCompletion = () => {
+                const allCompleted = operations.every(op => 
+                    op.state === ServiceLocationOperation.STATES.COMPLETED || 
+                    op.state === ServiceLocationOperation.STATES.CANCELLED
+                );
```

Comment:
this is reinventing callbacks (async.forEach) or promises (Promise.all) : code will be much simpler and efficient (no timer) if we just trigger each location's transition in parallel...